podcast/2.17.md

```diff
@@ -0,0 +1,487 @@
+
+I won't get too deep into what it is, but it had this mathematical component to it because it had to do with voting. There were these election algorithms that were involved in this product that he was building. He had this idea for exploring these new ways of collective decision-making through online voting. And he was taking time off to explore this idea and when we were hanging out, he said to me, "Oh, Max, you do physics. You're into reading mathematical stuff. Maybe you would be interested in researching this literature with me. This literature is about election theory and different algorithms for computing election results. We could code together."
+
+And it was so funny because at the time I was figuring out what I was going to do after college and was thinking about applying to material science grad school and things like that. And so I almost had this reaction: "Oh no, I don't have time to do this. I need to figure out what I want to do with my life." And luckily, I stopped and was like, "Oh, well, maybe this will help me figure out what I want to do." He shared a GitHub repository with me–it was a new thing to me at the time. This was 2011, I think. I was still in college.
+
+So I began to read about these election algorithms. He was developing an application that implemented these algorithms in Ruby and so I was looking up how to program in Ruby and how to adapt these algorithms into Ruby code. Eventually, we talked more and he said, "Oh, wow. You're really engaging with this. Well, maybe you and I should code together and I should come and visit you."
+
+I lived in Humboldt, California at the time and he lived in San Francisco, so he came and visited me. I remember we pair-programmed. That was something that he was really into at the time and I had never heard of it, but he set it all up. We had two displays and two keyboards, and all this equipment. We sat down and we coded and we ended up spending all this time pair-programming together.
+
+For me, my experience programming had been in the engineering department at school using very primitive tools. I was using VI–the engineering lab teachers taught us to use VI, not VIM, but VI in the terminal. And so Nathan was like, "We are coding Ruby." He had this Rails app, there was JavaScript, there was long polling, there was like Postgres–all this stuff. I was sitting next to him and taking it all in through this pair-programming process. I remember I was learning so much with all these technologies and tools for programming that I hadn't been exposed to. We would take turns typing, mostly he would type, and I would be taking it all in because there was so much, but I would type.
```

Comment:
```suggestion
For me, my experience programming had been in the engineering department at school using very primitive tools. I was using vi–the engineering lab teachers taught us to use vi, not Vim, but vi in the terminal. And so Nathan was like, "We are coding Ruby." He had this Rails app, there was JavaScript, there was long polling, there was like Postgres–all this stuff. I was sitting next to him and taking it all in through this pair-programming process. I remember I was learning so much with all these technologies and tools for programming that I hadn't been exposed to. We would take turns typing, mostly he would type, and I would be taking it all in because there was so much, but I would type.
```